api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,418 @@
+
+	// MachineNetwork is the list of IP address pools for machines.
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+
+	// NetworkType is the Container Network Interface (CNI) plug-in to install
+	// The default value is OpenShiftSDN for IPv4, and OVNKubernetes for IPv6 or SNO
+	// +kubebuilder:validation:Enum=OpenShiftSDN;OVNKubernetes
+	// +kubebuilder:default:=OVNKubernetes
+	// +optional
+	NetworkType string `json:"networkType,omitempty"`
+
+	// Additional cluster-wide annotations to be applied to the rendered templates
+	// +optional
+	CrAnnotations CrAnnotations `json:"crAnnotations,omitempty"`
+
+	// ClusterLabels is used to assign labels to the cluster to assist with policy binding.
+	// +optional
+	ClusterLabels map[string]string `json:"clusterLabels,omitempty"`
+
+	// InstallConfigOverrides is a generic way of passing install-config parameters.
+	// +optional
+	InstallConfigOverrides string `json:"installConfigOverrides,omitempty"`
+
+	// Json formatted string containing the user overrides for the initial ignition config
+	// +optional
+	IgnitionConfigOverride string `json:"ignitionConfigOverride,omitempty"`
+
+	// DiskEncryption is the configuration to enable/disable disk encryption for cluster nodes.
+	// +optional
+	DiskEncryption DiskEncryption `json:"diskEncryption,omitempty"`
+
+	// ProxySettings defines the proxy settings used for the install config
+	// +optional
+	ProxySettings aiv1beta1.Proxy `json:"proxy,omitempty"`
+
+	// ExtraManifestsRefs is list of config map references containing additional manifests to be applied to the cluster.
+	// +optional
+	ExtraManifestsRefs []corev1.LocalObjectReference `json:"extraManifestsRef,omitempty"`
+
+	// CPUPartitioning determines if a cluster should be setup for CPU workload partitioning at install time.
+	// When this field is set the cluster will be flagged for CPU Partitioning allowing users to segregate workloads to
+	// specific CPU Sets. This does not make any decisions on workloads it only configures the nodes to allow CPU Partitioning.
+	// The "AllNodes" value will setup all nodes for CPU Partitioning, the default is "None".
+	// +kubebuilder:validation:Enum=None;AllNodes
+	// +kubebuilder:default=None
+	// +optional
+	CPUPartitioning CPUPartitioningMode `json:"cpuPartitioningMode,omitempty"`
+
+	// TODO: is this field still needed?
+	// +optional
+	SiteConfigMap SiteConfigMap `json:"siteConfigMap,omitempty"`
+
+	// ControlPlaneAgents is the number of control-plane agents.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default:=1
+	ControlPlaneAgents uint8 `json:"controlPlaneAgents,omitempty"`
```

Comment:
These fields are referenced by the `AgentClusterInstall` manifest - we can perhaps remove them from here and introduce them as special variables wherein we compute the value: `.SpecialVars.ControlPlaneAgents` and `.SpecialVars.WorkerAgents`?

Reply:
@carbonin are there cases where clusters would be deployed where you did not want these values set to the defined number of control/worker nodes?

Reply:
These are used to determine when the correct number of hosts have been discovered. This is the trigger for install to start. So it should always be the number of hosts you will use to install.

Reply:
In the case of cluster expansion where a node is added to an existing cluster, is this field still relevant, or is this applicable only to initial installation? In the expansion case would changing this field (even if not relevant) be an error/incorrect? I'm thinking that it may be possible to always derive this value from the SiteConfig and not include it in the API. That would reduce one potential source of error which would block install (if number of hosts in the SiteConfig did not align with this value).

Reply:
Removed the `ControlPlaneAgents` and `WorkerAgents` fields from the SiteConfig API as it was agreed in a discussion that we can compute these values and introduce them as special variables for the AgentClusterInstall template manifest. This allows for less user prone errors.